modules/shared/src/main/scala/org/tessellation/security/Hashed.scala

```diff
@@ -1,13 +1,28 @@
 package org.tessellation.security
+
+import cats.kernel.Order
+import cats.syntax.contravariant._
+
 import org.tessellation.security.hash.{Hash, ProofsHash}
 import org.tessellation.security.signature.Signed
 
-import derevo.cats.{order, show}
-import derevo.derive
-
-@derive(order, show)
-case class Hashed[A <: AnyRef](signed: Signed[A], hash: Hash, proofsHash: ProofsHash)
+case class Hashed[+A <: AnyRef](signed: Signed[A], hash: Hash, proofsHash: ProofsHash)
 
 object Hashed {
   implicit def autoUnwrap[T <: AnyRef](t: Hashed[T]): T = t.signed
+
+  implicit def order[A <: AnyRef: Order]: Order[Hashed[A]] = Order.fromOrdering(ordering(Order[A].toOrdering))
+
+  implicit def ordering[A <: AnyRef: Ordering]: Ordering[Hashed[A]] = new HashedOrdering[A]()
+}
+
+final class HashedOrdering[A <: AnyRef: Ordering] extends Ordering[Hashed[A]] {
+
+  def compare(x: Hashed[A], y: Hashed[A]): Int =
+    Order
+      .whenEqual(
+        Order.fromOrdering(Ordering[A]).contramap[Hashed[A]](h => h.signed),
+        Order[Hash].contramap[Hashed[A]](h => h.hash)
```

Comment:
what about `ProofsHash`? I guess it should also serve as the field for defining order.

Reply:
`Order.whenEqual(x, y)` tries to order by `x` and then (if two elements are equal) falls back to `y` as a tiebreak. Hash is always a unique value so there is no need to combine another order 🤔 
@tbekas can you take a look and confirm what I wrote, or am I wrong?